processor/transformprocessor/internal/traces/processor_test.go

```diff
@@ -55,6 +55,17 @@ func Test_ProcessTraces_ResourceContext(t *testing.T) {
 			want: func(td ptrace.Traces) {
 			},
 		},
+		{
+			statement: `drop()`,
+			want: func(td ptrace.Traces) {
+				empty := ptrace.NewResourceSpansSlice()
+				empty.CopyTo(td.ResourceSpans())
+			},
+		},
+		{
+			statement: `drop() where dropped_attributes_count == 100`,
+			want:      func(td ptrace.Traces) {},
+		},
```

Comment:
If I understand this right, this is the case for a `where` clause not matching and leaving the `ptrace.Traces` unchanged.
Is it worth adding a case where some of the some of the events in a trace are dropped and others left in?